src/sync/incomingSyncUtils.js

```diff
@@ -306,9 +306,32 @@ export const createOrUpdateRecord = (database, settings, recordType, record) =>
   let internalRecord;
   switch (recordType) {
     case 'IndicatorAttribute': {
+      internalRecord = {
+        id: record.ID,
+        indicator: database.getOrCreate('ProgramIndicator', record.indicator_ID),
+        description: record.description,
+        code: record.code,
+        index: parseNumber(record.index),
+        isRequired: parseBoolean(record.is_required),
+        valueType: record.data_type && record.data_type.value,
+        valueDefault: record.data_type && record.data_type.default,
+        axis: record.axis,
+        isActive: parseBoolean(record.is_active),
+      };
+      database.update(recordType, internalRecord);
       break;
     }
     case 'IndicatorValue': {
+      const recordValue = record.value && JSON.parse(record.value);
+      internalRecord = {
+        id: record.ID,
+        facilityId: record.facility_ID,
+        period: database.getOrCreate('Period', record.period_ID),
```

Comment:
I think this will fail if you try to create a `Period` as it needs a `PeriodSchedule` - not sure the best move here.. seems like you need to create some placeholder `Period`, otherwise you won't be able to link up later.. but you can't create the `PeriodSchedule`. Hmm, no suggestion just yet!

Reply:
As discussed, simplest solution is to make `Period.periodSchedule` optional. Will just hotfix for now, but feature branch can be patched with any period logic needed to guard against regressions.

Reply:
Fixed by 0e7e1e2b.